### PR TITLE
add SplitSlice2Chunks utility function

### PR DESCRIPTION
**Pull request by avrahams**

### CODE REVIEW

File: httputils/httphelpers.go [hunk 1]
```diff
@@ -86,3 +88,68 @@ func HttpRespToString(resp *http.Response) (string, error) {
 
 	return respStr, err
 }
+
+//SplitSlice2Chunks - *recursively* splits a slice to chunks of sub slices that do not exceed max bytes size
+//Chunks might be bigger than max size if the slice contains element(s) that are bigger than the max size
+//this split algorithm fits for slices with elements that share more or less the same size per element
+//uses optimistic average size splitting to enhance performance and reduce the use of json encoding for size calculations
+func SplitSlice2Chunks[T any](slice []T, maxSize int, chunks chan<- []T, wg *sync.WaitGroup) {
```
Comment (Bezbran) [inline httputils/httphelpers.go:96]: add unitest

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/armosec/utils-go
 
-go 1.17
+go 1.18
```
Comment (Bezbran) [inline go.mod:3]: update GH actions

**Review by Bezbran — CHANGES_REQUESTED** [review]

File: httputils/httphelpers.go (hunk 1 above)
Comment (avrahams) [reply httputils/httphelpers.go:96]: done

File: go.mod (hunk 2 above)
Comment (avrahams) [reply go.mod:3]: done

**Review by Bezbran — APPROVED** [review]

**Pull request merged**